addons/project/controllers/portal.py

```diff
@@ -146,20 +90,35 @@ def portal_my_projects(self, page=1, date_begin=None, date_end=None, sortby=None
         })
         return request.render("project.portal_my_projects", values)
 
-    @http.route(['/my/project/<int:project_id>'], type='http', auth="public", website=True)
-    def portal_my_project(self, project_id=None, access_token=None, page=1, date_begin=None, date_end=None, sortby=None, search=None, search_in='content', groupby=None, **kw):
+    @http.route(['/my/project/<int:project_id>',
+                 '/my/project/<int:project_id>/page/<int:page>',
+                 '/my/project/<int:project_id>/task/<int:task_id>',
+                 '/my/project/<int:project_id>/project_sharing',
+                 '/my/task',
+                 '/my/task/page/<int:page>'], type='http', auth="public")
+    def portal_project_routes_outdated(self, **kwargs):
+        """Redirect the outdated routes to the new routes."""
+        path = request.httprequest.path
+        if path.startswith('/my/project/'):
+            return request.redirect(path.replace('/my/project/', '/my/projects/'))
```

Comment:
Might be worth two routes no ? :D

Reply:
2 methods? One for project and one for task? 🙂 
Why not, it is the same for me. ^^

Reply:
Yep, but do as you like :D

Reply:
Ok, thanks for your reviews! 😉